clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1904,6 +1905,68 @@ public void testUnsubscribeWithoutGroupId() {
         verify(applicationEventHandler).add(ArgumentMatchers.isA(UnsubscribeEvent.class));
     }
 
+    @Test
+    public void testSeekToBeginning() {
+        SubscriptionState subscriptions = mock(SubscriptionState.class);
+        Collection<TopicPartition> topics = Collections.singleton(new TopicPartition("test", 0));
+        consumer = newConsumer(
+                mock(FetchBuffer.class),
+                new ConsumerInterceptors<>(Collections.emptyList()),
+                mock(ConsumerRebalanceListenerInvoker.class),
+                subscriptions,
+                "group-id",
+                "client-id");
+        completeResetOffsetEventSuccessfully();
+        consumer.seekToBeginning(topics);
+        verify(subscriptions).requestOffsetReset(topics, OffsetResetStrategy.EARLIEST);
```

Comment:
I'm not sure why we need to verify what we've mocked. `requestOffsetReset` is executed because we call it when mocking `applicationEventHandler#addAndGet`. That means this test can pass even if `process(ResetOffsetEvent)` doesn't call `requestOffsetReset`, right?

Perhaps we should verify the content of `ResetOffsetEvent` instead. The behavior of `ApplicationEventProcessor.process(ResetOffsetEvent)` is already covered by `ApplicationEventProcessorTest#testResetOffsetEvent`.

Reply:
> we are testing the integration between AsyncConsumer and ApplicationEventProcessor.

`AsyncConsumerTest` doesn't cover this scenario since the 'behavior' of `ApplicationEventHandler` is mocked. This isn't the actual behavior, so 'verifying' the 'mocked' behavior is meaningless. It's like writing a test to test the test code.